src/unix/darwin.c

```diff
@@ -280,14 +280,15 @@ static int uv__get_cpu_speed(uint64_t* speed) {
                                                     NULL,
                                                     0);
         if (freq_ref) {
-          uint32_t freq;
+          const UInt8* freq_ref_ptr = pCFDataGetBytePtr(freq_ref);
           CFIndex len = pCFDataGetLength(freq_ref);
-          CFRange range;
-          range.location = 0;
-          range.length = len;
+          if (len == 8)
+            *speed = *(const uint64_t*) freq_ref_ptr;
```

Comment:
Is the data always aligned on an 8 byte boundary? I'd use `memcpy()` to play it safe.

Reply:
I think we should stick with the cast strategy. In the unlikely case that data is unaligned we would take an insignificant performance hit (this is macOS/iOS only code). If we go to memcpy then we have to consider endianness for the 4 byte case

Reply:
The compiler will always turn memcpy into an efficient (but safe) load

Reply:
Since there is no ill effect with the current implementation, and since a memcpy solution is more complicated, I suggest sticking with the current proposal.

Reply:
That is where I am confused. All architectures that this code will run on can handle the (unlikely) case that the date is unaligned & as such I don't think the code is invalid.
If this is a coding style issue, then I can update it.

Reply:
Many common architectures don't support this (for example ARM / AArch64: https://blog.quarkslab.com/unaligned-accesses-in-cc-what-why-and-solutions-to-do-it-properly.html)

Reply:
The world isn't just x86 and amd64

Reply:
I have changed the impl to use memcpy